### PR TITLE
Remove deprecated features

### PR DESCRIPTION
**Pull request by angriman**
This PR removes the methods that were deprecated in #421. `Graph::nodes/edges` are handled in #462 because they are replaced by new iterators for nodes and edges, which should be provided before `Graph::nodes/edges` are removed.

### CODE REVIEW

**Comment by avdgrinten** [comment]
This should wait for one release cycle.

**Comment by avdgrinten** [comment]
The new iterators *should* be split to a new PR so that we can merge them in the current release cycle.

**Comment by angriman** [comment]
The iterators commits have been moved to #462.

**Comment by avdgrinten** [comment]
Did you verify that everything that is removed in this PR has been deprecated for two major releases?

**Comment by angriman** [comment]
The recursive algorithm for SCCs has been deprecated in NetworKit 6.0, all the remaining removed features have been deprecated in NetworKit <= 5.0.

File: networkit/__init__.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def getComponentPartition(G):
 			cc = components.ConnectedComponents(G).run()
 		return cc.getPartition()
 
-	print("Network Properties for:\t\t{}".format(G.getName()))
+	print("Network Properties:")
```
Comment (fabratu) [inline networkit/__init__.py:137]: Is G.getName() dropped completely?

Comment (angriman) [reply networkit/__init__.py:137]: Yes, I think that line is still file, the remaining network properties will be listed below.

File: include/networkit/generators/DynamicGraphSource.hpp [hunk 2]
```diff
@@ -74,7 +76,7 @@ class DynamicGraphSource {
     /**
      * Continue generating until the number of time steps reaches this upper limit.
      */
-    virtual void generateTimeSteps(count t);
+    virtual void TLX_DEPRECATED(generateTimeSteps(count t));
```
Comment (avdgrinten) [inline include/networkit/generators/DynamicGraphSource.hpp:79]: Why is this deprecated now?

File: input/example2.dgs [hunk 3]
```diff
@@ -3,7 +3,6 @@ authorGraph 0 0
 an 1
 an 2
 ae 1-2 1 2 weight=1
-st 1
```
Comment (avdgrinten) [inline input/example2.dgs:6]: Huh?

File: networkit/GEXFIO.py [hunk 4]
```diff
@@ -105,15 +105,10 @@ def read(self, fpath):
 
 		#5.2 Sorting the eventStream by time and adding timeStep between events that happen in different times
 		self.eventStream.sort(key=lambda x:x[1])
-		for i in range(1, len(self.eventStream)):
-			if self.eventStream[i][1] != self.eventStream[i-1][1]:
-				self.eventStream.append((GraphEvent(GraphEvent.TIME_STEP, 0, 0, 0), self.eventStream[i-1][1]))
-		self.eventStream.sort(key=lambda x:x[1])
```
Comment (avdgrinten) [inline networkit/GEXFIO.py:111]: Why is this removed?

File: networkit/cpp/dynamics/DGSStreamParser.cpp [hunk 5]
```diff
@@ -58,9 +58,7 @@ std::vector<GraphEvent> DGSStreamParser::getStream() {
             std::string tag = split[0];
 
             // parse commands
-            if (tag.compare("st") == 0) { // clock
-                stream.emplace_back(GraphEvent::TIME_STEP);
-            } else if (tag.compare("an") == 0) { // add node
+            if (tag.compare("an") == 0) { // add node
```
Comment (avdgrinten) [inline networkit/cpp/dynamics/DGSStreamParser.cpp:61]: Why are these changes done?

File: networkit/cpp/dynamics/DGSStreamParser.cpp [hunk 6]
```diff
@@ -120,9 +118,7 @@ std::vector<GraphEvent> DGSStreamParser::getStream() {
...
```
Comment (avdgrinten) [inline networkit/cpp/dynamics/DGSStreamParser.cpp:121]: Why are these changes done?

**Review by avdgrinten — COMMENTED** [review]
There are several changes related to time steps. We should have a replacement available for the functionality that is removed here.

**Comment by angriman** [comment]
Those changes are done because timestamps are not supported anymore as already discussed. What replacement are you referring to?

**Comment by avdgrinten** [comment]
That they are not supported in `Graph` anymore does not mean that we have to drop support entirely. What are the timestamps good for / what is the use case?

**Comment by angriman** [comment]
Time steps are not used by any algorithm in NetworKit so far but I see the point of not dropping support entirely as they might be useful for dynamic graphs. However, I don't see an easy way to do that as time steps are graph events and thus inherently tied to `Graph`. If we move time steps from `Graph` how should they be managed?

**Comment by avdgrinten** [comment]
If they are not used in any algorithm, it might be fine to remove the feature. However, support in graph readers etc. should be kept.

**Comment by hmeyerhenke** [comment]
Time stamps were meant to be used in dynamic algorithms, yes. I think in the end we realized the dynamic algorithms realized as part of NetworKit in a different way. Not sure if that is true for user code, though (but if nobody complained about deprecation...). Keeping support in graph readers still makes sense, I guess.

File: include/networkit/generators/DynamicGraphSource.hpp (hunk 2 above)
Comment (angriman) [reply include/networkit/generators/DynamicGraphSource.hpp:79]: I forgot to deprecate this before, the implementation is this
```
void DynamicGraphSource::generateTimeSteps(count t) {
     while (G->time() < t) {
         this->generate();
     }
     G->shrinkToFit();
  }
```
which is something that we cannot support anymore if `time` is removed from `Graph`.

**Comment by michitux** [comment]
I have implemented a dynamic graph generator in NetworKit (see https://github.com/michitux/networkit/tree/ckbdynamic) that uses time steps. The generator has the concept that in each time step a set of events happens. After each time step, a community detection algorithm should detect a given (also dynamic) reference community structure. I therefore use time steps to mark the point where the communities should be detectable, and the idea for plots would be that you have these time steps on one axis. None of my code relies on the graph knowing which time step it represents, but it certainly relies on time step events.

**Comment by angriman** [comment]
Alright, so I removed the time steps only from `Graph`, but I kept the support for time step events.

File: include/networkit/generators/DynamicGraphSource.hpp (hunk 2 above)
Comment (fabratu) [reply include/networkit/generators/DynamicGraphSource.hpp:79]: I think given the total amount of changes, it is totally fine to miss something - but IMHO I would remove the function directly. As of now it does nothing than compacting the graph, which is misleading.

Comment (angriman) [reply include/networkit/generators/DynamicGraphSource.hpp:79]: That's true, OTOH removing it might break some other people's code so probably deprecating it is the safest solution.

**Comment by avdgrinten** [comment]
Sounds reasonable. Maybe timestamps would fit well into a framework for arbitrary data ("attributes") attached to graphs/nodes/edges (as people suggested support for such a feature at ND last week).

Maybe the easiest path forward is to split the time step stuff into a different PR and to merge the remainder of the PR independently.

**Comment by angriman** [comment]
Agree, I dropped the time steps commit.

File: include/networkit/generators/DynamicGraphSource.hpp (hunk 2 above)
Comment (fabratu) [reply include/networkit/generators/DynamicGraphSource.hpp]: Even though it is not interesting for this PR, but (a more general opinion):

I would prefer breaking code over unexpected behavior (and you don't know what caused it). Or at least mention here, that the function might not do what you think.

**Review by avdgrinten — APPROVED** [review]
LGTM.
